game/factions/faction.py

```diff
@@ -0,0 +1,216 @@
+
+        faction.frontline_units = [f for f in [vehicle_loader(vehicle) for vehicle in json.get("frontline_units", [])] if f is not None]
+        faction.artillery_units = [f for f in [vehicle_loader(vehicle) for vehicle in json.get("artillery_units", [])] if f is not None]
+        faction.infantry_units = [f for f in [vehicle_loader(vehicle) for vehicle in json.get("infantry_units", [])] if f is not None]
+        faction.logistics_units = [f for f in [vehicle_loader(vehicle) for vehicle in json.get("logistics_units", [])] if f is not None]
+
+        faction.sams = json.get("sams", [])
+        faction.shorads = json.get("shorads", [])
+        faction.missiles = json.get("missiles", [])
+        faction.name = json.get("requirements", {})
+
+        faction.carrier_names = json.get("carrier_names", [])
+        faction.helicopter_carrier_names = json.get("helicopter_carrier_names", [])
+        faction.navy_generators = json.get("navy_generators", [])
+        faction.aircraft_carrier = [f for f in [ship_loader(vehicle) for vehicle in json.get("aircraft_carrier", [])] if f is not None]
+        faction.helicopter_carrier = [f for f in [ship_loader(vehicle) for vehicle in json.get("helicopter_carrier", [])] if f is not None]
+        faction.destroyers = [f for f in [ship_loader(vehicle) for vehicle in json.get("destroyers", [])] if f is not None]
+        faction.cruisers = [f for f in [ship_loader(vehicle) for vehicle in json.get("cruisers", [])] if f is not None]
+        faction.has_jtac = json.get("has_jtac", False)
+        faction.jtac_unit = aircraft_loader(json.get("jtac_unit", None))
+        faction.navy_group_count = int(json.get("navy_group_count", 1))
+        faction.missiles_group_count = int(json.get("missiles_group_count", 0))
+
+        # Load doctrine
+        doctrine = json.get("doctrine", "modern")
+        if doctrine == "modern":
+            faction.doctrine = MODERN_DOCTRINE
+        elif doctrine == "coldwar":
+            faction.doctrine = COLDWAR_DOCTRINE
+        elif doctrine == "ww2":
+            faction.doctrine = WWII_DOCTRINE
+        else:
+            faction.doctrine = MODERN_DOCTRINE
+
+        # Load the building set
+        building_set = json.get("building_set", "default")
+        if building_set == "default":
+            faction.building_set = DEFAULT_AVAILABLE_BUILDINGS
+        elif building_set == "ww2ally":
+            faction.building_set = WW2_ALLIES_BUILDINGS
+        elif building_set == "ww2germany":
+            faction.building_set = WW2_GERMANY_BUILDINGS
+        else:
+            faction.building_set = DEFAULT_AVAILABLE_BUILDINGS
+
+        return faction
+
+    @property
+    def units(self):
+        return self.infantry_units + self.aircrafts + self.awacs + self.artillery_units + self.frontline_units + self.tankers + self.logistics_units
+
+
+def unit_loader(unit: str, class_repository:[]) -> Optional[PlaneType]:
```

Comment:
Does mypy like this? I think you need to specify the list contents (I always use the explicit `List` from `typing`, so not sure).